tests/unit/amo/components/TestInstallWarning.js

```diff
@@ -0,0 +1,333 @@
+import * as React from 'react';
+
+import InstallWarning, {
+  EXPERIMENT_CATEGORY_DISPLAY,
+  EXPERIMENT_ID,
+  INSTALL_WARNING_EXPERIMENT_DIMENSION,
+  VARIANT_INCLUDE_WARNING,
+  VARIANT_EXCLUDE_WARNING,
+  InstallWarningBase,
+} from 'amo/components/InstallWarning';
+import { setInstallState } from 'core/actions/installations';
+import {
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_STATIC_THEME,
+  INSTALLED,
+  UNINSTALLED,
+  UNKNOWN,
+} from 'core/constants';
+import { createInternalAddon } from 'core/reducers/addons';
+import {
+  createFakeTracking,
+  dispatchClientMetadata,
+  fakeAddon,
+  fakeI18n,
+  fakeInstalledAddon,
+  getFakeLogger,
+  shallowUntilTarget,
+  userAgentsByPlatform,
+} from 'tests/unit/helpers';
+import Notice from 'ui/components/Notice';
+
+describe(__filename, () => {
+  let store;
+
+  beforeEach(() => {
+    store = dispatchClientMetadata({
+      userAgent: userAgentsByPlatform.mac.firefox57,
```

Comment:
Is this relevant?

Reply:
Well, it's not strictly needed because `dispatchClientMetadata` defaults to a user agent of Firefox, but I figured, as the code specifically expects the user agent to be Firefox, then it would be best to explicitly set that in the test, rather than just rely on the default value set by `dispatchClientMetadata`.